nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  python = pkgs.python3Packages.python;
+  cfg = config.services.seafile;
+
+  ccnetConf = pkgs.writeText "ccnet.conf" ''
+    [General]
+    SERVICE_URL = ${cfg.serviceUrl}
+  '';
+
+  seafileConf = pkgs.writeText "seafile.conf" ''
```

Comment:
I think it would be beneficial to provide a `settings` like option instead. See [RFC42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md) for details, and feel free to ping me if you need a hand.